src/core/net/private/geoipdatabase.cpp

```diff
@@ -214,402 +168,330 @@ QString GeoIPDatabase::lookup(const QHostAddress &hostAddr) const
 #else
     Q_IPV6ADDR addr = hostAddr.toIPv6Address();
 #endif
-    const quint32 nodeCount = static_cast<quint32>(m_geoIPData->index.size());
-    Node node = m_geoIPData->index[0];
+
+    static const int nodeSize = m_recordSize / 4; // in bytes
+    static const int indexSize = m_nodeCount * nodeSize;
+    static const int recordBytes = nodeSize / 2;
```

Comment:
Change these 3 consts to member variables and in initialize them when you parsing metadata.